lti_consumer/lti_consumer.py

```diff
@@ -0,0 +1,839 @@
+
+Resources
+---------
+
+Background and detailed LTI specification can be found at:
+
+    http://www.imsglobal.org/specs/ltiv1p1p1/implementation-guide
+
+This module is based on the version 1.1.1 of the LTI specification by the
+IMS Global authority. For authentication, it uses OAuth1.
+
+When responding back to the LTI tool provider, we must issue a correct
+response. Types of responses and their message payload is available at:
+
+    Table A1.2 Interpretation of the 'CodeMajor/severity' matrix.
+    http://www.imsglobal.org/gws/gwsv1p0/imsgws_wsdlBindv1p0.html
+
+A resource to test the LTI protocol (PHP realization):
+
+    http://www.imsglobal.org/developers/LTI/test/v1p1/lms.php
+
+We have also begun to add support for LTI 1.2/2.0.  We will keep this
+docstring in synch with what support is available.  The first LTI 2.0
+feature to be supported is the REST API results service, see specification
+at
+http://www.imsglobal.org/lti/ltiv2p0/uml/purl.imsglobal.org/vocab/lis/v2/outcomes/Result/service.html
+
+What is supported:
+------------------
+
+1.) Display of simple LTI in iframe or a new window.
+2.) Multiple LTI components on a single page.
+3.) The use of multiple LTI providers per course.
+4.) Use of advanced LTI component that provides back a grade.
+    A) LTI 1.1.1 XML endpoint
+        a.) The LTI provider sends back a grade to a specified URL.
+        b.) Currently only action "update" is supported. "Read", and "delete"
+            actions initially weren't required.
+    B) LTI 2.0 Result Service JSON REST endpoint
+       (http://www.imsglobal.org/lti/ltiv2p0/uml/purl.imsglobal.org/vocab/lis/v2/outcomes/Result/service.html)
+        a.) Discovery of all such LTI http endpoints for a course.  External tools GET from this discovery
+            endpoint and receive URLs for interacting with individual grading units.
+            (see lms/djangoapps/courseware/views.py:get_course_lti_endpoints)
+        b.) GET, PUT and DELETE in LTI Result JSON binding
+            (http://www.imsglobal.org/lti/ltiv2p0/mediatype/application/vnd/ims/lis/v2/result+json/index.html)
+            for a provider to synchronize grades into edx-platform.  Reading, Setting, and Deleteing
+            Numeric grades between 0 and 1 and text + basic HTML feedback comments are supported, via
+            GET / PUT / DELETE HTTP methods respectively
+"""
+
+import logging
+import bleach
+import re
+import json
+import urllib
+
+from datetime import datetime
+from collections import namedtuple
+from webob import Response
+
+from xblock.core import String, Scope, List, XBlock
+from xblock.fields import Boolean, Float, Integer
+from xblock.fragment import Fragment
+
+from xblockutils.resources import ResourceLoader
+from xblockutils.studio_editable import StudioEditableXBlockMixin
+
+from .exceptions import LtiError
+from .oauth import log_authorization_header
+from .lti import Lti1Consumer, Lti2Consumer
+from .outcomes import OutcomeService
```

Comment:
I'd argue that explicit relative imports are safer, since they can't ever introduce a second way of importing the module; given that this will be installed as a separate package, there will be only a single way of importing the modules anyway, so there is no risk of anything going wrong regardless of what types of import is used here. :)

That said, I can't see a reason why `from lti_consumer.exceptions import LtiError` shouldn't work.  @douglashall could you post the full traceback of the ImportError?